app/src/main/java/ch/epfl/cs311/wanderwave/viewmodel/MapViewModel.kt

```diff
@@ -43,6 +52,21 @@ constructor(val locationSource: LocationSource, private val beaconRepository: Be
 
   private val _areBeaconsLoaded = MutableStateFlow(false)
 
+  private val _profile =
+      MutableStateFlow(
+          Profile(
+              firstName = "My FirstName",
+              lastName = "My LastName",
+              description = "My Description",
+              numberOfLikes = 0,
+              isPublic = true,
+              spotifyUid = "My Spotify UID",
+              firebaseUid = "My Firebase UID",
+              profilePictureUri = null))
+  val profile: StateFlow<Profile> = _profile
+  private var _profileUI = MutableStateFlow(ProfileViewModel.UIState())
+  val profileUI: StateFlow<ProfileViewModel.UIState> = _profileUI
```

Comment:
Are the public states `profile` and `profileUI` necessary? From what I can see, `profileUI` is never used, and `profile` only in a test...

Reply:
Also, I don't think both of the private states are necessary, right? UiState is just a profile with a loading and error parameter attached, but we don't need that here. I think you should be able to just use one flow.